### PR TITLE
MR-751 Add archived check to bc

### PR DESCRIPTION
**Pull request by Asura5**
- Some messages are received by the bonuscalc listener for operatives that are "archived" - in other words, they are marked as having left Hackney
- Clearly, if they are receiving "work completed" notifications, they are still working for Hackney. Nobody comes in to work for free.
- Therefore, if they've been archived, but they are getting SQS messages, unarchive them with haste

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ services:
       - test-database
 
   test-database:
-    image: postgres:12
+    image: postgres:12.13
```
Comment (ange47rm) [inline docker-compose.yml:53]: What prompted you to update this version?

File: BonusCalcListener.Tests/Gateway/TimesheetGatewayTests.cs [hunk 2]
```diff
@@ -41,118 +41,14 @@ public async Task RetrievesCorrectTimesheetIfClosedInOpenWeek()
         public async Task RetrievesCorrectTimesheetIfClosedInClosedWeek()
         {
             // Arrange
-            await SeedData();
+            await SeedData().ConfigureAwait(false);
```
Comment (ange47rm) [inline BonusCalcListener.Tests/Gateway/TimesheetGatewayTests.cs:44]: I see data is now seeded in DatabaseTests.cs, why do we still seed data in this test, unlike the test above?

File: BonusCalcListener.Tests/Gateway/OperativesGatewayTests.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+        [SetUp]
+        public void Setup()
+        {
+            _loggerMock = new Mock<ILogger<OperativesGateway>>();
+            _classUnderTest = new OperativesGateway(BonusCalcContext, _loggerMock.Object);
+        }
+
+        [Test]
+        public async Task RetrievesOperativeIfOperativeFound()
```
Comment (ange47rm) [inline BonusCalcListener.Tests/Gateway/OperativesGatewayTests.cs:30]: I would suggest changing this title to ActivatesOperativeIfArchivedOperativeFound

File: BonusCalcListener.Tests/Gateway/TimesheetGatewayTests.cs (hunk 2 above)
Comment (Asura5) [reply BonusCalcListener.Tests/Gateway/TimesheetGatewayTests.cs]: Fixed

File: BonusCalcListener.Tests/Gateway/OperativesGatewayTests.cs (hunk 3 above)
Comment (Asura5) [reply BonusCalcListener.Tests/Gateway/OperativesGatewayTests.cs]: renamed

**Review by ange47rm — APPROVED** [review]

**Pull request merged**